ansible/roles/java-base/tasks/main.yml

```diff
@@ -29,13 +29,26 @@
 
 # if this fails you want to check in vars/main.yml and add package name
 # as appropriate -- try to use generic os family if available.
+
+- name: Check if java is already installed
+  raw: java -version
+  register: java
+  ignore_errors: yes
+
 - name: install java
-  when: not os|startswith("zos")
+  when: java.rc > 0 and not os|startswith("zos") and not os|startswith("macos")
   package: name="{{ java_package_name }}" state=present
 
 - name: install webupd8 oracle java 8 extras
-  when: os in ("ubuntu1204", "ubuntu1404")
+  when: java.rc > 0 and os in ("ubuntu1204", "ubuntu1404")
   package: name="{{item}}" state=present
   with_items:
     - ca-certificates
     - oracle-java8-set-default
+
+- name: install java (macos)
+  become_user: administrator
```

Comment:
I think the become_user can be set in one place for macos as opposed to in individual jobs

Reply:
There are certain places where we wan't to run jobs as root on macOS. The main issue is that brew commands cannot be run as root (hence why I dropped the privileges down for all brew commands)